frontend/src/app/components/modals/export-modal/wp-table-export.modal.ts

```diff
@@ -1,23 +1,40 @@
-import {ChangeDetectorRef, Component, ElementRef, Inject, OnInit} from '@angular/core';
+import {ChangeDetectorRef, Component, ElementRef, Inject, OnInit, ViewChild, OnDestroy} from '@angular/core';
 import {OpModalLocalsMap} from 'core-components/op-modals/op-modal.types';
 import {WorkPackageViewColumnsService} from 'core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-columns.service';
 import {OpModalComponent} from 'core-components/op-modals/op-modal.component';
 import {IsolatedQuerySpace} from "core-app/modules/work_packages/query-space/isolated-query-space";
-import {UrlParamsHelperService} from "core-components/wp-query/url-params-helper";
 import {WorkPackageCollectionResource} from "core-app/modules/hal/resources/wp-collection-resource";
 import {HalLink} from "core-app/modules/hal/hal-link/hal-link";
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 import {OpModalLocalsToken} from "core-components/op-modals/op-modal.service";
 import * as URI from 'urijs';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import {LoadingIndicatorService} from "core-app/modules/common/loading-indicator/loading-indicator.service";
+import { expand, concatMap, catchError } from 'rxjs/operators';
+import { timer, Subscription } from 'rxjs';
 
 interface ExportLink extends HalLink {
   identifier:string;
 }
 
+/*
+Modal for exporting work packages to different formats. The user may choose from a variety of formats (e.g. PDF and CSV).
+
+The backend may choose to provide the export right away (synchronously) or delayed. In the later case, the modal will poll the
+backend until the export is done.
+Because the modal has to deal with both cases, without knowing upfront whether the download will be delayed or not, it needs
+to treat every download expecting it to be delayed. Because of this, the modal cannot simply provide download hrefs which would
+ allow the browser to download the export but rather has to first check whether an export is delayed or not, and if it is delayed,
+ it has to wait until the export is ready. Because of the necessary check, the modal has a hidden link that it clicks on to perform
+ the actual download once the export is ready (delayed or not).
+
+The modal might also be used to only display the progress of an export. This will happen if a link for exporting is provided via the locals.
+ */
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/5cc0b80f0d93730810e407445c446a54cf5a577c/frontend/src/app/components/modals/export-modal/wp-table-export.modal.ts#L32"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>